### PR TITLE
[LinalgExt] Switch tiling LinalgExt tests to use transform dialect.

### PR DESCRIPTION
**Pull request by hanhanW**
This allows us to test tiling LinalgExt through interfaces. In this context, we don't need to implement its own tiling passes.

The lit tests change because

1. The upstream interface methods extract the destination slices from iter_args, not original tensor.
2. We have better helpers that use OpFoldResult (i.e., `makeComposedFoldedAffineApply`) in upstream methods.
3. Delete the lit test about tiling on reduction dims.

### CODE REVIEW

**Comment by hanhanW** [comment]
There are few major workarounds in the commit:

If we tile an op with size=0, it still create extract_slice op for the whole tensor. And the extract_slice op is not folded away.
We don't signal an error when tiling happens on reduction dims. I think we can remove such test because they are configurable by users.
There is a type mismatch in scan_1d test. It generates tensor<?xi32> when tile_size=0 while the original shape is tensor<128xi32>. The tile size mismatch in this case, so I disable the test.
I will try to fix (1) and (3); delete the test about (2).

I verified that https://github.com/llvm/llvm-project/pull/75410 address (1) and (3). We can remove the tests about (2) and switch the tests to use transform dialect.

File: llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/tiling.mlir [hunk 1]
```diff
@@ -15,8 +14,13 @@ func.func @scatter_tiling(
     } -> tensor<?x?xf32>
   return %0 : tensor<?x?xf32>
 }
-//   CHECK-DAG: #[[MAP0:.+]] = affine_map<(d0)[s0, s1] -> (10, -d0 + s1)>
-//   CHECK-DAG: #[[MAP1:.+]] = affine_map<(d0)[s0, s1] -> (20, -d0 + s1)>
+transform.sequence failures(propagate) {
+^bb1(%module_op: !transform.any_op):
+  %0 = transform.structured.match ops{["iree_linalg_ext.scatter"]} in %module_op : (!transform.any_op) -> !transform.any_op
+  %1, %loops:2 = transform.structured.tile_using_for %0 [10, 20] : (!transform.any_op) -> (!transform.any_op, !transform.any_op, !transform.any_op)
```
Comment (hanhanW) [inline llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/tiling.mlir:20]: I verified that it is using `scf::tileUsingSCFForOp` method.

https://github.com/llvm/llvm-project/blob/e775ba384efe31928bac796b2a4d388a1c298c5e/mlir/lib/Dialect/Linalg/TransformOps/LinalgTransformOps.cpp#L2660-L2661

File: llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/tiling.mlir [hunk 2]
```diff
@@ -29,24 +33,24 @@ func.func @scatter_tiling(
...
-//       CHECK:       %[[USED_TILESIZEX:.+]] = affine.min #[[MAP1]](%[[IV1]])[%[[TILESIZEX]], %[[D1]]]
+//       CHECK:       %[[USED_TILESIZEX:.+]] = affine.min #[[MAP1]](%[[IV1]])[%[[D1]]]
 //       CHECK:       %[[UPDATE_SLICE:.+]] = tensor.extract_slice %[[UPDATES]][%[[IV0]], %[[IV1]]]
 //  CHECK-SAME:           [%[[USED_TILESIZEY]], %[[USED_TILESIZEX]]]
 //       CHECK:       %[[INDEX_SLICE:.+]] = tensor.extract_slice %[[INDICES]][%[[IV0]], 0]
 //  CHECK-SAME:           [%[[USED_TILESIZEY]], 1]
-//       CHECK:       %[[SCATTER_DIM:.+]] = tensor.dim %[[ORIGINAL]], %[[C0]]
-//       CHECK:       %[[ORIGINAL_SLICE:.+]] = tensor.extract_slice %[[ORIGINAL]][0, %[[IV1]]]
+//       CHECK:       %[[SCATTER_DIM:.+]] = tensor.dim %[[INITX]], %[[C0]]
+//       CHECK:       %[[ORIGINAL_SLICE:.+]] = tensor.extract_slice %[[INITX]][0, %[[IV1]]]
```
Comment (MaheshRavishankar) [inline llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/tiling.mlir:45]: This is changing the semantics a bit... just want to make sure this is expected.

**Review by MaheshRavishankar — APPROVED** [review]

File: llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/tiling.mlir (hunk 2 above)
Comment (hanhanW) [reply llvm-external-projects/iree-dialects/test/Dialect/iree_linalg_ext/tiling.mlir:45]: Yes, this is expected and it is what we're doing in IREE (i.e., they both use upstream scf methods).

**Pull request merged**